web/shared-beans/src/main/java/io/deephaven/web/shared/data/CustomColumnDescriptor.java

```diff
@@ -95,9 +92,15 @@ public static boolean isCompatible(List<CustomColumnDescriptor> was, List<Custom
     }
 
     public static List<CustomColumnDescriptor> from(String[] newCustomColumns) {
+        final Set<String> descriptorNames = new HashSet<>();
         final List<CustomColumnDescriptor> list = new ArrayList<>();
         for (String col : newCustomColumns) {
-            list.add(new CustomColumnDescriptor().setExpression(col));
+            CustomColumnDescriptor descriptor = new CustomColumnDescriptor().setExpression(col);
+            if (descriptorNames.contains(descriptor.getName())) {
```

Comment:
you could put the `descriptorNames.add(descriptor.getName())` in here instead - Set.add returns a boolean, true if the set didn't contain the item already:

```
            if (!descriptorNames.add(descriptor.getName())) {
```
and then not add it below